File: example3/src/main/java/javatest/colorhaake/com/example3/network/RetrofitUtil.java
package javatest.colorhaake.com.example3.network;

import javatest.colorhaake.com.example3.BuildConfig;
import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;

public class RetrofitUtil {
    public static OkHttpClient genericClient() {

        HttpLoggingInterceptor interceptor = new HttpLoggingInterceptor();
        interceptor.setLevel(
                BuildConfig.DEBUG ? HttpLoggingInterceptor.Level.BODY : HttpLoggingInterceptor.Level.NONE);

        return new OkHttpClient.Builder()
                .addInterceptor(interceptor)
                .build();
    }
}

File: example3/src/main/java/javatest/colorhaake/com/example3/network/RetrofitUtil.kt
package javatest.colorhaake.com.example3.network

import javatest.colorhaake.com.example3.BuildConfig
import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor

object RetrofitUtil {
    fun genericClient(): OkHttpClient {

        val interceptor = HttpLoggingInterceptor()
        interceptor.level =
                if (BuildConfig.DEBUG) HttpLoggingInterceptor.Level.BODY
                else HttpLoggingInterceptor.Level.NONE

        return OkHttpClient.Builder()
                .addInterceptor(interceptor)
                .build()
    }
}
